### PR TITLE
adding sql file for HW2

### PR DESCRIPTION
**Pull request by eesalt**
works off of baseball database. calculates batting average for several scenarios and stores in tables for easy access. Explores several methods for calculating batting averages. See notes for performance frustrations.

### CODE REVIEW

File: HW2/HW2.sql [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+#ALTER TABLE inning MODIFY COLUMN game_id INT UNSIGNED NOT NULL;
+
+CREATE INDEX game_id ON batter_counts(game_id);
+
+#assignment
+#Calculate the batting average using SQL queries for every player
+CREATE TABLE batter_ba as
+SELECT a.batter,
+       CASE WHEN sum(a.atBat)>0 THEN
+       format(sum(a.Hit) / sum(a.atBat),3)
```
Comment (matherdy) [inline HW2/HW2.sql]: Maybe use round here instead.  Not sure if format rounds the number or just cuts it off

File: HW2/HW2.sql [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+       year(b.local_date)        as YEAR,
+       CASE WHEN sum(a.atBat) > 0 THEN
+       format(sum(a.Hit) / sum(a.atBat),3)
+           ELSE NULL END as Batting_Average
+from batter_counts a,
+     game b
+WHERE a.game_id = b.game_id
+group by a.batter,
+         year
+order by 1, 2;
```
Comment (matherdy) [inline HW2/HW2.sql:30]: using numbers here is a little ambiguous.  Might be easier to read if you use the names of the columns instead or leave a comment.

File: HW2/HW2.sql [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+# 100 rolling days
+# this assumes that the game_ids are in sequential order.
+# Normally I wouldn't do that but the performance was so bad I had to
+# See below for my preferred method
+CREATE TABLE batter_100_day_ba as
+SELECT a.game_id,
+       a.batter,
+       CASE WHEN sum(c.atBat) > 0 THEN
+       format(sum(c.Hit) / sum(c.atBat), 3)
```
Comment (matherdy) [inline HW2/HW2.sql]: Same here maybe use round

File: HW2/HW2.sql [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+           ELSE NULL END as 100_Day_Batting_Average
```
Comment (matherdy) [inline HW2/HW2.sql:44]: maybe use more descriptive nick names for the tables so its easier to read

File: HW2/HW2.sql (hunk 2 above)
Comment (dafrenchyman) [reply HW2/HW2.sql:30]: Agreed!

**Pull request merged**